tox.ini

```diff
@@ -0,0 +1,43 @@
+[tox]
+basepython = python3
+recreate = true
+skipsdist = true
+envlist =
+    flake8
+    pylint
+    shellcheck
+
+[common]
+pyfiles =
+    bin/ec2metadata
+    bin/write-mime-multipart
+shfiles =
+    bin/cloud-localds
+    bin/cloud-publish-ubuntu
+    bin/growpart
+    bin/mount-image-callback
+    bin/resize-part-image
+    bin/ubuntu-cloudimg-query
+    bin/vcs-run
+
+[testenv:flake8]
+deps = flake8==3.8.3
+commands = python3 -m flake8 {[common]pyfiles}
+
+[testenv:pylint]
+deps = pylint==2.5.3
+commands = python3 -m pylint {[common]pyfiles}
+
+[testenv:flake8-tip]
+deps = flake8
+commands = python3 -m flake8 {[common]pyfiles}
+
+[testenv:pylint-tip]
+deps = pylint
+commands = python3 -m pylint {[common]pyfiles}
+
+[testenv:shellcheck]
+# 'shellcheck' is not Python, but tox is a usable wrapper to run any kind of
+# tests. Let's use it for common test-run as well for unification purposes.
```

Comment:
This comment is a little misleading; we are installing a Python package and invoking a Python command in the virtualenv, so as far as tox is concerned it _is_ Python.

(I'm only pointing this out because my initial assumption was that I would need shellcheck installed for this to work, and this comment lead me further down that mistaken path.)

Reply:
And you are right! When I wrote the comment I was using shellcheck from the Ubuntu package, but in Xenial it's too old and it doesn't support the `--severity` argument. I'll clarify the comment.

The alternative to `py-shellcheck` are:
 - Running a job just for shellcheck on a newer Ubuntu version. A bit of a waste of resources and more wait time to get the test results for little gain.
 - Fixing all the shellcheck nits in the scripts :)

The `py-shellcheck` project is small but looks well maintained.

What's your opinion on using it?

Reply:
Comment clarified, but I'm still interested on your take on shellcheck-py vs. other solutions.